pkg/client/didexchange/client.go

```diff
@@ -48,18 +49,39 @@ func New(ctx provider) (*Client, error) {
 }
 
 // CreateInvitation create invitation
-func (c *Client) CreateInvitation() (*InvitationRequest, error) {
+func (c *Client) CreateInvitation() (*didexchange.Invitation, error) {
 	verKey, err := c.wallet.CreateKey()
 	if err != nil {
 		return nil, fmt.Errorf("failed CreateSigningKey: %w", err)
 	}
 
-	return &InvitationRequest{Invitation: &didexchange.Invitation{
+	return &didexchange.Invitation{
 		ID:              uuid.New().String(),
 		Label:           "agent", // TODO pass label as argument
 		RecipientKeys:   []string{verKey},
 		ServiceEndpoint: c.inboundTransportEndpoint,
-	}}, nil
+		Type:            didexchange.ConnectionInvite,
+	}, nil
+}
+
+// HandleInvitation handle incoming invitation
+func (c *Client) HandleInvitation(payload []byte) error {
```

Comment:
If someone happens to have an Invitation struct rather than []byte, I guess we are currently expecting them to Marshal prior to calling HandleInvitation.

I wonder if this is the right pattern.

Reply:
I'm not sure which better []byte or struct

Reply:
if found in the future struct is better I will change it

Reply:
I generally tend to think struct.

Reply:
@llorllale @rolsonquadras

Reply:
probably be best to accept the struct itself

Reply:
I change it to struct